src/collective/feedback/restapi/services/add.py

```
@@ -48,18 +50,32 @@ def validate_form(self, form_data):
         """
         check all required fields and parameters
         """
-        for field in ["vote"]:
+        for field in ["vote", "content"]:
             value = form_data.get(field, "")
             if not value:
                 raise BadRequest("Campo obbligatorio mancante: {}".format(field))
 
+    def looks_like_path(self, string):
+        return bool(re.match(r"^(/|/[^\s<>:\"|?*]+.*)$", string))
+
     def extract_data(self, form_data):
-        context_state = api.content.get_view(
-            context=self.context,
-            request=self.request,
-            name="plone_context_state",
-        )
-        context = context_state.canonical_object()
-        form_data.update({"uid": context.UID()})
-        form_data.update({"title": context.Title()})
+        path = form_data.pop("content")
+        if self.looks_like_path(path):
+            portal = api.portal.get()
+            contextual_path = "/" + portal.id + path
+            context = api.content.get(path=contextual_path)
```

What if context is none? context.UID() will raise an exception.

Probably if context is none, we should also check allowed_views: if it's not in that mapping, raise an exception.

---

Done

At this point, I prefer to raise an exception instead of checking the allowed_views list, because if it were a valid view, there shouldn't have been any entry in this if.